src/foam/nanos/approval/ApprovableAware.js

```diff
@@ -55,27 +57,41 @@
                 logger.error("Error instantiating : ", obj.getClass().getSimpleName(), e);
               }
               Map diff = oldObj == null ? null : oldObj.diff(obj);
+              StringBuilder hash_sb = new StringBuilder(obj.getClass().getSimpleName());
+              if ( operation == Operations.UPDATE && obj instanceof ApprovableAware ) 
+                hash_sb.append(((ApprovableAware) obj).getStringId());
+
               if ( diff != null ) {
                 // remove ids, timestamps and userfeedback
-                diff.remove("id");
+                if ( operation == Operations.CREATE ) diff.remove("id");
                 diff.remove("created");
                 diff.remove("lastModified");
                 diff.remove("userFeedback");
+
                 // convert array properties to list to get consistent hash
+                // and create hash
                 Iterator it = diff.entrySet().iterator();
-                List<String> arrayProps = new ArrayList<String>();
+
                 while( it.hasNext() ) {
                   Map.Entry next = (Map.Entry) it.next();
-                  if ( next.getValue() instanceof Object[] ) {
-                    arrayProps.add((String) next.getKey());
+                  Object nextValue = next.getValue();
+                  if ( nextValue instanceof Object[] ) {
+                    next.setValue(Arrays.asList((Object[]) nextValue));
                   }
-                }
-                for ( String prop : arrayProps ) {
-                  diff.put(prop, Arrays.asList((Object[]) diff.get(prop)).hashCode());
+                  if ( nextValue instanceof FObject ) {
+                    String hashedKey = getApprovableHashKey(x, (FObject) nextValue, Operations.CREATE);
```

Comment:
Why is this hardcoded to Operation.CREATE?
```suggestion
                    String hashedKey = getApprovableHashKey(x, (FObject) nextValue, operation);
```